File: Programmers/src/Javascript/안전지대.js
function solution(board) {
    let dx = [-1,0,1,1,1,0,-1,-1];
    let dy = [-1,-1,-1,0,1,1,1,0];
    
    for(let i=0; i<board.length; i++){
        for(let j=0; j<board.length; j++) {
            if(board[i][j]===1){
                for(let k=0; k<8; k++){
                    let nx = i + dx[k];
                    let ny = j + dy[k];
                    
                    if(0<=ny && ny<board.length && 0<=nx && nx<board.length && board[nx][ny]===0)
                        board[nx][ny]=2;
                }
            }
        }
    }
    
    return board.reduce((acc, e) => acc + e.filter(el => el === 0).length, 0);

}

File: Programmers/src/Java/Lv0/안전지대.java
class Solution {
    static int[] dx = {-1,0,1,0,-1,1,-1,1};
    static int[] dy = {0,1,0,-1,-1,-1,1,1};
    public int solution(int[][] board) {
        int answer = 0;
        boolean[][] visited = new boolean[board.length][board.length];
        
        for(int i=0; i<board.length; i++) {
            for(int j=0; j<board.length; j++) {
                for(int k=0; k<8; k++) {
                    int nx = i + dx[k];
                    int ny = j + dy[k];
                    
                    if(board[i][j]==1)  visited[i][j] = true;
                    if(nx>=0 && nx<board.length && ny>=0 && ny<board.length && board[nx][ny]==1) {
                        visited[i][j] = true;
                    }
                }
            }
        }
        
        for(int i=0; i<visited.length; i++) {
            for(int j=0; j<visited.length; j++) {
                if(!visited[i][j])
                    answer++;
            }
        }
        
        return answer;
    }
}